src/Nethermind/Nethermind.State/StateProvider.cs

```diff
@@ -761,12 +764,15 @@ private void IncrementChangePosition()
             Resettable<Change>.IncrementPosition(ref _changes, ref _capacity, ref _currentPosition);
         }
 
-        private void SetupCache(Address address)
+        private Stack<int> SetupCache(Address address)
         {
-            if (!_intraBlockCache.ContainsKey(address))
+            ref Stack<int>? value = ref _intraBlockCache.GetValueRefOrAddDefault(address, out bool exists);
+            if (!exists)
             {
-                _intraBlockCache[address] = new Stack<int>();
+                value = new Stack<int>();
```

Comment:
this new dictionary is not added to _intraBlockCache

Reply:
It is as `value` is a `ref`